contrib/jemalloc-cmake/CMakeLists.txt

```diff
@@ -17,7 +17,11 @@ if (ENABLE_JEMALLOC)
         #
         # By enabling percpu_arena number of arenas limited to number of CPUs and hence
         # this problem should go away.
-        set (JEMALLOC_CONFIG_MALLOC_CONF "percpu_arena:percpu,oversize_threshold:0")
+        #
+        # muzzy_decay_ms -- use MADV_FREE when available on newer Linuxes, to avoid
+        # spurios latencies and additional work associated with MADV_DONTNEED.
+        # See https://github.com/ClickHouse/ClickHouse/issues/11121 for motivation.
+        set (JEMALLOC_CONFIG_MALLOC_CONF "percpu_arena:percpu,oversize_threshold:0,muzzy_decay_ms:10000")
```

Comment:
AFAICS this is the default value, that said that now it should be the same as in upstream

Reply:
Where can I find it to check? I was looking here: https://github.com/jemalloc/jemalloc/blob/ea6b3e973b477b8061e0076bb257dbd7f3faa756/include/jemalloc/internal/arena_types.h#L12 and also seen a comment in changelog about disabling muzzy decay by default.

Reply:
>Where can I find it to check? 

You are right, my bad, I got this from ["documentation"](http://jemalloc.net/jemalloc.3.html#opt.muzzy_decay_ms) (I knew that I should look into sources instead, sigh)

So, jemalloc/jemalloc@8e9a613122251d4c519059f8e1e11f27f6572b4c "disables" it by default, but according to documentation:

>A decay time of 0 causes all unused muzzy pages to be purged immediately upon creation. A decay time of -1 disables purging. 

So with default settings it should purge (free) muzzy pages immediately, and after first version of this PR it uses mmap for this, maybe it slows down free, but speedsup allocation then, and hence latency goes down?

>muzzy_decay_ms -- use MADV_FREE when available on newer Linuxes

From the documentation:

>Approximate time in milliseconds from the creation of a set of unused muzzy pages until an equivalent set of unused muzzy pages is purged (i.e. converted to clean) and/or reused. Muzzy pages are defined as previously having been unused dirty pages that were subsequently purged in a manner that left them subject to the reclamation whims of the operating system (e.g. madvise(...MADV_FREE)), and therefore in an indeterminate state. 

So AFAICS `muzzy_decay_ms` is muzzy->clean, so it uses `madvice(DONTNEED)` or `mmap`, `dirty_decay_ms` uses `MADV_FREE`